src/app/InteractionModelEngine.h

```diff
@@ -164,6 +164,17 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate
 
 void DispatchSingleClusterCommand(chip::ClusterId aClusterId, chip::CommandId aCommandId, chip::EndpointId aEndPointId,
                                   chip::TLV::TLVReader & aReader, Command * apCommandObj);
+
+/**
+ *  Check if a cluster command exists, the caller would expect this function to look up a catalog to find if the cluster / command
+ * exists on the endpoint.
+ *  TODO: The implementation lives in ember-compatibility-functions.cpp, this should be replaced by IM command catalog look up
+ * function after we have a cluster catalog in interaction model engine.
+ *
+ *  @retval  #CHIP_NO_ERROR There is a command on cluster on the endpoint.
+ *  @retval  #CHIP_ERROR_INVALID_PROFILE_ID The command or cluster not exists on the endpoint.
```

Comment:
Why is this function not just returning a boolean?

Reply:
Yes, we can return a boolean value here.